poly-synth/src/poly-synth-demo.js

```diff
@@ -22,163 +22,368 @@ class PolySynthDemo {
         });
 
     let filterDetuneSlider_ = new ParamController(
-        filterADSRId, this.polySynth_.setParameter.bind(this.polySynth_), {
+        containerId, this.polySynth_.setParameter.bind(this.polySynth_), {
           name: 'Detune Amount',
           id: 'filterDetuneAmount',
           type: 'range',
           min: this.polySynth_.minDetuneAmount,
           max: this.polySynth_.maxDetuneAmount,
           step: 0.01,
-          default: this.polySynth_.getParameters().filterDetuneAmount,
+          default: this.polySynth_.getParameters().filterDetuneAmount
+        });
+  }
+  
+  initializeBitcrusherGUI_(containerId){
+    let bitcrusherBitDepthSlider_ = new ParamController(
+        containerId, this.polySynth_.setBitDepth.bind(this.polySynth_), {
+          name: 'Bit Depth',
+          id: 'bitDepth',
+          type: 'range',
+          min: this.polySynth_.minBitDepth,
+          max: this.polySynth_.maxBitDepth,
+          step: 0.01,
+          default: this.polySynth_.bitcrusher.bitDepth
         });
+    bitcrusherBitDepthSlider_.disable();
+
+    let bitcrusherReductionSlider_ = new ParamController(
+        containerId, this.polySynth_.setReduction.bind(this.polySynth_), {
+          name: 'Reduction',
+          id: 'reduction',
+          type: 'range',
+          min: this.polySynth_.minReduction,
+          max: this.polySynth_.maxReduction,
+          step: 1,
+          default: this.polySynth_.bitcrusher.reduction
+        });
+    bitcrusherReductionSlider_.disable();
+
+    // Sound is not processed by the bitcrusher if in bypass mode.
+    document.getElementById('bitcrusherBypassButton').onclick = (event) => {
+      // Only one of |this.polySynth_.activeBitcrusherRoute_| and
+      // |this.polySynth_.bypassBitcrusherRoute_| has a non-zero gain.
+      if (event.target.textContent === 'Active') {
+        event.target.textContent = 'Bypassed';
+
+        // The change is scheduled slightly into the future to avoid glitching.
+        const t = this.context_.currentTime + 0.01;
+        this.polySynth_.bypassBitcrusherRoute.gain.linearRampToValueAtTime(
+            1, t);
+        this.polySynth_.activeBitcrusherRoute.gain.linearRampToValueAtTime(
+            0, t);
+
+        bitcrusherBitDepthSlider_.disable();
+        bitcrusherReductionSlider_.disable();
+      } else {
+        event.target.textContent = 'Active';
+        const t = this.context_.currentTime + 0.01;
+        this.polySynth_.activeBitcrusherRoute.gain.linearRampToValueAtTime(
+            1, t);
+        this.polySynth_.bypassBitcrusherRoute.gain.linearRampToValueAtTime(
+            0, t);
+
+        bitcrusherBitDepthSlider_.enable();
+        bitcrusherReductionSlider_.enable();
+      }
+    }
+  }
+
+  initializeConvolverGUI_(containerId, selectorId) {
+    let convolverWetnessSlider_ = new ParamController(
+        containerId, this.polySynth_.setConvolverWetness.bind(this.polySynth_),
+        {
+          name: 'Wetness',
```

Comment:
It is `mix` parameter of `Reverb` component. So the bigger mix value means you get more reverberation. I think it's intuitive - also it is the convention.